Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/DefaultHoughFunctions.hpp

```diff
@@ -0,0 +1,116 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020-2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Utilities/Result.hpp"
+
+#pragma once
+
+enum class HoughError {
+   Failure = 1,
+      SomethingElse,
+      };
+std::error_code make_error_code(HoughError e) {
+   return {static_cast<int>(e), std::generic_category()};
+}
+
+namespace std {
+// register with STL
+   template <>
+   struct is_error_code_enum<HoughError> : std::true_type {};
+}  // namespace std
+
+namespace ActsExamples {
+   namespace DefaultHoughFunctions {
+      using ResultDouble = Acts::Result<double>;
+      using ResultBool = Acts::Result<bool>;
+      using ResultUnsigned = Acts::Result<unsigned>;
+            
+      ResultDouble fieldCorrectionDefault(unsigned region, double y, double r) {
+         if (region == 999)
+            return y + r;  // this should not be found, for now this is a dummy to show what one *could* do
+         return ResultDouble::success(0.0);
+      }
+      
+      ResultUnsigned findLayerIDDefault(double r) {
```

Comment:
To make this configurable, wouldn't that be a good place to use the `Delegate<>` from the Utilities, that comes with practically zero cost overhead and you can make this customisable.

Reply:
Please ignore, I saw that these are actually used in Delegates, so great thinking ...